src/cci/cas_cci.c

```diff
@@ -712,14 +713,24 @@ cci_disconnect (int mapped_conn_id, T_CCI_ERROR * err_buf)
 
       get_last_error (con_handle, err_buf);
     }
-  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool (con_handle->id) >= 0)
+  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT])
     {
-      cci_end_tran_internal (con_handle, CCI_TRAN_ROLLBACK);
-      API_ELOG (con_handle, 0);
+      if (con_handle->id < 0)
+	{
+	  return CCI_ER_INVALID_ARGS;
```

Comment:
cci_get_data () shows below code.
`  if (indicator == NULL || value == NULL)
    {
      return CCI_ER_INVALID_ARGS;
    }`

This 'indicator' variable is argument from cci_get_data ().
And then 'con_handle->id' is also parameter of hm_put_con_to_pool.
So, If 'con_handle->id' is negative value, it is INVALID_ARGS. So, I set the error above.

Reply:
When I insert -1 into **con_handle->id** it occurred core dump.
This protection code is needed.

Reply:
cci_get_data() This function is CCI API, hm_put_con_to_pool() This function is CCI internal function.
CCI_ER_INVALID_ARGS appears to be an error code that informs the user of an error when the parameter of the CCI API function is incorrect.
Since it is an error that occurred inside CCI, wouldn't it be appropriate to handle it with a different error code?

And, the part that assigns -1 to con_handle->id also seems to need modification.

Reply:
It is fixed. Would you check it again?